server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/data/StarTreeFileFormatsTests.java

```diff
@@ -0,0 +1,210 @@
+        Map<Long, InMemoryTreeNode> levelOrderStarTreeNodeMap = new LinkedHashMap<>();
+        InMemoryTreeNode root = generateSampleTree(levelOrderStarTreeNodeMap);
+        long starTreeDataLength = StarTreeWriter.writeStarTree(dataOut, root, 7, "star-tree");
+
+        // asserting on the actual length of the star tree data file
+        assertEquals(starTreeDataLength, 247);
+        dataOut.close();
+
+        dataIn = directory.openInput("star-tree-data", IOContext.READONCE);
+
+        StarTreeMetadata starTreeMetadata = mock(StarTreeMetadata.class);
+        when(starTreeMetadata.getDataLength()).thenReturn(starTreeDataLength);
+        when(starTreeMetadata.getDataStartFilePointer()).thenReturn(0L);
+        StarTree starTree = new StarTree(dataIn, starTreeMetadata);
+
+        StarTreeNode starTreeNode = starTree.getRoot();
+        Queue<StarTreeNode> queue = new ArrayDeque<>();
+        queue.add(starTreeNode);
+
+        while ((starTreeNode = queue.poll()) != null) {
+
+            // verify the star node
+            assertStarTreeNode(starTreeNode, levelOrderStarTreeNodeMap.get(starTreeNode.getDimensionValue()));
+
+            Iterator<? extends StarTreeNode> childrenIterator = starTreeNode.getChildrenIterator();
+
+            if (starTreeNode.getChildDimensionId() != -1) {
+                while (childrenIterator.hasNext()) {
+                    StarTreeNode child = childrenIterator.next();
+                    assertStarTreeNode(
+                        starTreeNode.getChildForDimensionValue(child.getDimensionValue(), false),
+                        levelOrderStarTreeNodeMap.get(child.getDimensionValue())
+                    );
+                    queue.add(child);
+                }
+            }
+        }
+
+        dataIn.close();
+
+    }
+
+    private void assertStarTreeNode(StarTreeNode starTreeNode, InMemoryTreeNode treeNode) throws IOException {
+        assertEquals(starTreeNode.getDimensionId(), treeNode.dimensionId);
+        assertEquals(starTreeNode.getDimensionValue(), treeNode.dimensionValue);
+        assertEquals(starTreeNode.getStartDocId(), treeNode.startDocId);
+        assertEquals(starTreeNode.getEndDocId(), treeNode.endDocId);
+        assertEquals(starTreeNode.getChildDimensionId(), treeNode.childDimensionId);
+        assertEquals(starTreeNode.getAggregatedDocId(), treeNode.aggregatedDocId);
+
+        if (starTreeNode.getChildDimensionId() != -1) {
+            assertFalse(starTreeNode.isLeaf());
+            if (treeNode.children != null) {
+                assertEquals(starTreeNode.getNumChildren(), treeNode.children.values().size());
+            }
+        } else {
+            assertTrue(starTreeNode.isLeaf());
+        }
+
+    }
+
+    private InMemoryTreeNode generateSampleTree(Map<Long, InMemoryTreeNode> levelOrderStarTreeNode) {
+        // Create the root node
+        InMemoryTreeNode root = new InMemoryTreeNode();
```

Comment:
Why not randomize number of dimensions , number of nodes for each dimension, start doc id , end doc id etc for each node. Just we need to ensure dimensionid and childDimension id are as per the levels they're in. 

We can have good amount of nodes upto 500 in that case and more than 5-6 levels in star tree. 

The randomness as per this comment - https://github.com/opensearch-project/OpenSearch/pull/15295/files#r1724783961 - we'll use binary search of FixedLengthStarTreeNode vs Java binary search based on custom comparator of in memory node maybe.